### PR TITLE
[Refactor] Refactor fileio

### PR DESCRIPTION
**Pull request by xiexinch**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Use the new fileio from mmengine
https://github.com/open-mmlab/mmengine/pull/533

## Modification

1. Use `mmengine.fileio` to repalce FileClient  in mmseg/datasets
2. Use `mmengine.fileio` to repalce FileClient in mmseg/datasets/transforms
3. Use `mmengine.fileio` to repalce FileClient in mmseg/visualization

## BC-breaking (Optional)

TODO

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by MeowZheng — COMMENTED** [review]
please remove the modifications of `projects` and add the version of mmcv and mmengine that `backend_args` need.

File: mmseg/__init__.py [hunk 1]
```diff
@@ -58,17 +58,17 @@ def digit_version(version_str: str, length: int = 4):
...
 mmengine_min_version = digit_version(MMENGINE_MIN)
 mmengine_max_version = digit_version(MMENGINE_MAX)
 mmengine_version = digit_version(mmengine.__version__)
 
-assert (mmengine_min_version <= mmengine_version < mmengine_max_version), \
+assert (mmengine_min_version <= mmengine_version <= mmengine_max_version), \
     f'MMEngine=={mmengine.__version__} is used but incompatible. ' \
     f'Please install mmengine>={mmengine_min_version}, '\
-    f'<{mmengine_max_version}.'
+    f'<={mmengine_max_version}.'
```
Comment (MeowZheng) [inline mmseg/__init__.py:72]:
```suggestion
    f'<{mmengine_max_version}.'
```

File: mmseg/__init__.py [hunk 2]
```diff
@@ -58,17 +58,17 @@ def digit_version(version_str: str, length: int = 4):
...
     f'MMCV=={mmcv.__version__} is used but incompatible. ' \
-    f'Please install mmcv==2.0.0rc3.'
+    f'Please install mmcv>=2.0.0rc4.'
 
 mmengine_min_version = digit_version(MMENGINE_MIN)
 mmengine_max_version = digit_version(MMENGINE_MAX)
 mmengine_version = digit_version(mmengine.__version__)
 
-assert (mmengine_min_version <= mmengine_version < mmengine_max_version), \
+assert (mmengine_min_version <= mmengine_version <= mmengine_max_version), \
```
Comment (MeowZheng) [inline mmseg/__init__.py:69]:
```suggestion
assert (mmengine_min_version <= mmengine_version < mmengine_max_version), \
```

File: mmseg/datasets/transforms/loading.py [hunk 3]
```diff
@@ -202,23 +203,22 @@ class LoadBiomedicalImageFromFile(BaseTransform):
         to_float32 (bool): Whether to convert the loaded image to a float32
             numpy array. If set to False, the loaded image is an float64 array.
             Defaults to True.
-        file_client_args (dict): Arguments to instantiate a FileClient.
-            See :class:`mmengine.fileio.FileClient` for details.
-            Defaults to ``dict(backend='disk')``.
+        backend_args (dict): Arguments to instantiate a file backend.
+            See https://mmengine.readthedocs.io/en/latest/api/fileio.htm
+            for details. Defaults to ``dict(backend='local')``
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/loading.py:208]: Also add `Notes: mmcv>=2.0.0rc4, mmengine>=0.2.0 required.`

**Review by MeowZheng — APPROVED** [review]

**Review by MengzhangLI — APPROVED** [review]

File: docs/en/notes/faq.md [hunk 4]
```diff
@@ -4,37 +4,19 @@ We list some common troubles faced by many users and their corresponding solutio
...
-|         0.9.0          |   mmcv-full>=1.1.4, \<1.3.0    |            Not required             |          Not required          |
-|         0.8.0          |   mmcv-full>=1.1.4, \<1.2.0    |            Not required             |          Not required          |
-|         0.7.0          |   mmcv-full>=1.1.2, \<1.2.0    |            Not required             |          Not required          |
-|         0.6.0          |   mmcv-full>=1.1.2, \<1.2.0    |            Not required             |          Not required          |
+| MMSegmentation version |          MMCV version          | MMEngine version  | MMClassification (optional) version | MMDetection (optional) version |
+| :--------------------: | :----------------------------: | :---------------: | :---------------------------------: | :----------------------------: |
+|     dev-1.x branch     |        mmcv >= 2.0.0rc4        | MMEngine >= 0.2.0 |           mmcls>=1.0.0rc0           | mmdet>=3.0.0rc4, \<=3.0.0rc5>  |
+|       1.x branch       |        mmcv >= 2.0.0rc4        | MMEngine >= 0.2.0 |           mmcls>=1.0.0rc0           | mmdet>=3.0.0rc4, \<=3.0.0rc5>  |
```
Comment (MeowZheng) [inline docs/en/notes/faq.md:12]:
```suggestion
|       1.x branch       |        mmcv == 2.0.0rc3        | MMEngine >= 0.1.0 |           mmcls>=1.0.0rc0           | mmdet>=3.0.0rc4, \<=3.0.0rc5>  |
```
as 1.x branch  updates  when mmseg release, I suggest modifying 1.x branch dependency in bump pr

File: mmseg/__init__.py [hunk 5]
```diff
@@ -7,8 +7,8 @@
 
 from .version import __version__, version_info
 
-MMCV_MIN = '2.0.0rc3'
-MMCV_MAX = '2.0.0rc3'
+MMCV_MIN = '2.0.0rc4'
+MMCV_MAX = '2.1.0'
 MMENGINE_MIN = '0.1.0'
```
Comment (MeowZheng) [inline mmseg/__init__.py:12]:
```suggestion
MMCV_MAX = '2.1.0'
MMENGINE_MIN = '0.2.0'
```

File: requirements/mminstall.txt [hunk 6]
```diff
@@ -1,4 +1,4 @@
 mmcls>=1.0.0rc0
-mmcv==2.0.0rc3
+mmcv>=2.0.0rc4
 mmdet==3.0.0rc5
 mmengine>=0.1.0,<1.0.0
```
Comment (MeowZheng) [inline requirements/mminstall.txt:4]:
```suggestion
mmengine>=0.2.0,<1.0.0
```

File: docs/en/notes/faq.md (hunk 4 above)
Comment (xiexinch) [reply docs/en/notes/faq.md:12]: > as 1.x branch updates when mmseg release, I suggest modifying 1.x branch dependency in bump pr

get

**Review by MeowZheng — APPROVED** [review]

**Pull request merged**